File: backend/app/schemas/match.py
from datetime import datetime

from pydantic import BaseModel, Field


class JobMatchScore(BaseModel):
    overall_score: int = Field(ge=0, le=100)
    strong: list[str] = Field(default_factory=list)
    missing: list[str] = Field(default_factory=list)
    summary: str | None = None


class JobMatchResult(JobMatchScore):
    job_id: int
    computed_at: datetime


class JobMatchRead(BaseModel):
    computed: bool
    match: JobMatchResult | None = None

File: frontend/src/types/match.ts
export type JobMatchResult = {
  job_id: number;
  overall_score: number;
  strong: string[];
  missing: string[];
  summary: string | null;
  computed_at: string;
};

export type JobMatchRead = {
  computed: boolean;
  match: JobMatchResult | null;
};

export function matchScoreTier(score: number): "high" | "medium" | "low" {
  if (score >= 80) return "high";
  if (score >= 60) return "medium";
  return "low";
}

export function formatMatchDate(iso: string): string {
  return new Date(iso).toLocaleString(undefined, {
    dateStyle: "medium",
    timeStyle: "short",
  });
}
